### PR TITLE
Add acceptance tests for debug_traceTransaction endpoint

### PR DESCRIPTION
**Pull request by anastasiya-kovaliova**
**Description**:
This PR adds acceptance tests to cover the behaviour of the debug_traceTransaction in both positive and negative scenarious:

- Adds positive and negative acceptance tests
- Adds tests brief description in the design document

**Related issue(s):**

**Fixes** 
#1792 

Notes for reviewer:
The ticket for implementing the debug_traceTransaction is: #1477 

Checklist

- [x]  Documented (Code comments, README, etc.)
- [x]  Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/server/tests/helpers/assertions.ts [hunk 1]
```diff
@@ -398,4 +399,25 @@ export default class Assertions {
     expect(address).to.match(/(\b0x[a-f0-9]{40}\b)/g, 'matches evm address format');
     expect(address).to.match(/(\b0x(0){15})/g, 'contains 15 consecutive zeros');
   };
+
+  static validateResultDebugValues = (result, excludedValues, nestedExcludedValues, expectedResult) => {
+    let hasValidHash = (currentValue) => RelayAssertions.validateHash(currentValue);
+
+    //Validate result schema
```
Comment (Nana-EC) [inline packages/server/tests/helpers/assertions.ts:406]: nits:
```suggestion
    // Validate result schema
```

File: packages/server/tests/acceptance/rpc_batch3.spec.ts [hunk 2]
```diff
@@ -1056,4 +1073,818 @@ describe('@api-batch-3 RPC Server Acceptance Tests', function () {
...
+        ...defaultTransactionFields,
+        type: 0,
+      };
+
+      transactionType2930 = {
+        ...defaultTransactionFields,
+        type: 1,
+      };
+
+      transactionType2 = {
```
Comment (Nana-EC) [inline packages/server/tests/acceptance/rpc_batch3.spec.ts:1117]: use `...defaultTransactionFields` as you did in the above

File: packages/server/tests/acceptance/rpc_batch3.spec.ts [hunk 3]
```diff
@@ -1056,4 +1073,818 @@ describe('@api-batch-3 RPC Server Acceptance Tests', function () {
...
+        gasLimit: defaultGasLimit,
+        type: 2,
+        maxFeePerGas: defaultGasPrice,
+        maxPriorityFeePerGas: defaultGasPrice,
+      };
+
+      //deploy estimate gas contract
+      const transaction = {
+        ...transactionTypeLegacy,
+        chainId: 0x12a,
```
Comment (Nana-EC) [inline packages/server/tests/acceptance/rpc_batch3.spec.ts:1130]: Q: should already be captured by `transactionTypeLegacy` right?

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Looking good.
minor suggestions.
Also please add `@release` in one or two of the happy path tests so they get picked up with release testing.

**Review by Nana-EC — APPROVED** [review]

File: packages/server/tests/acceptance/rpc_batch3.spec.ts [hunk 4]
```diff
@@ -1056,4 +1073,809 @@ describe('@api-batch-3 RPC Server Acceptance Tests', function () {
...
+      };
+
+      transactionTypeLegacy = {
+        ...defaultTransactionFields,
+        type: 0,
+      };
+
+      transactionType2930 = {
+        ...defaultTransactionFields,
+        type: 1,
```
Comment (konstantinabl) [inline packages/server/tests/acceptance/rpc_batch3.spec.ts:1114]: @anastasiya-kovaliova TransactionType2930 should also have field accessList: []

File: packages/server/tests/acceptance/rpc_batch3.spec.ts [hunk 5]
```diff
@@ -1056,4 +1073,809 @@ describe('@api-batch-3 RPC Server Acceptance Tests', function () {
...
+      };
+
+      transactionType2 = {
+        ...defaultTransactionFields,
+        chainId: Number(CHAIN_ID),
```
Comment (konstantinabl) [inline packages/server/tests/acceptance/rpc_batch3.spec.ts:1119]: @anastasiya-kovaliova Is there a reason we have chainId overwritten here?

File: packages/server/tests/acceptance/rpc_batch3.spec.ts [hunk 6]
```diff
@@ -1056,4 +1073,809 @@ describe('@api-batch-3 RPC Server Acceptance Tests', function () {
...
+        it('should be able to debug a successful CREATE transaction of type 2930 with call depth and onlyTopCall false', async function () {
+          const transaction = {
+            ...transactionType2930,
+            chainId: 0x12a,
+            data: bytecode,
+            nonce: await relay.getAccountNonce(accounts[0].address, requestId),
+            gasPrice: await relay.gasPrice(requestId),
+          };
+
+          let signedTransaction = await accounts[0].wallet.signTransaction(transaction);
```
Comment (konstantinabl) [inline packages/server/tests/acceptance/rpc_batch3.spec.ts:1417]: @anastasiya-kovaliova You can use const for the variables that do not get reassigned, like thos two : )) There were other places as well

File: packages/server/tests/helpers/assertions.ts [hunk 7]
```diff
@@ -398,4 +399,25 @@ export default class Assertions {
     expect(address).to.match(/(\b0x[a-f0-9]{40}\b)/g, 'matches evm address format');
     expect(address).to.match(/(\b0x(0){15})/g, 'contains 15 consecutive zeros');
   };
+
+  static validateResultDebugValues = (result, excludedValues, nestedExcludedValues, expectedResult) => {
+    let hasValidHash = (currentValue) => RelayAssertions.validateHash(currentValue);
```
Comment (konstantinabl) [inline packages/server/tests/helpers/assertions.ts:404]: @anastasiya-kovaliova Here you can use const as well

File: packages/server/tests/helpers/assertions.ts [hunk 8]
```diff
@@ -398,4 +399,25 @@ export default class Assertions {
     expect(address).to.match(/(\b0x[a-f0-9]{40}\b)/g, 'matches evm address format');
     expect(address).to.match(/(\b0x(0){15})/g, 'contains 15 consecutive zeros');
   };
+
+  static validateResultDebugValues = (result, excludedValues, nestedExcludedValues, expectedResult) => {
```
Comment (konstantinabl) [inline packages/server/tests/helpers/assertions.ts:403]: @anastasiya-kovaliova Please add types as well - for the parameters, the return value and the variables in the function. I know we do not have them in a lot of places, but we should start adding them

**Review by konstantinabl — CHANGES_REQUESTED** [review]

**Review by Nana-EC — APPROVED** [review]

**Review by AlfredoG87 — APPROVED** [review]
LGTM

**Review by konstantinabl — APPROVED** [review]

**Comment by konstantinabl** [comment]
LG

**Pull request merged**